lib/src/arithmetic/values/floating_point_values/floating_point_value.dart

```diff
@@ -492,19 +492,45 @@ class FloatingPointValue implements Comparable<FloatingPointValue> {
         return this;
       }
     } else if (subend.isAnInfinity) {
-      return subend.negate();
+      return subend.negated();
     } else if (isAnInfinity) {
       return this;
     }
     return _performOp(subend, (a, b) => a - b);
   }
 
   /// Negate operation for [FloatingPointValue].
-  FloatingPointValue negate() => clonePopulator().populate(
+  FloatingPointValue negated() => clonePopulator().populate(
       sign: sign.isZero ? LogicValue.one : LogicValue.zero,
       exponent: exponent,
       mantissa: mantissa);
 
+  // Move this to FixedPointValuePopulator.ofFloatingPointValue.
+  // The question is how to make FixedPoint a flexible output size?
+
+  // There could be two forms:  one that is value dependent widths
+  // and one that is fixed width to match the FPV capacity.
+
+  /// Losslessly convert a [FloatingPointValue] to a [FixedPointValue].
+  FixedPointValue toFixedPointValue() {
+    // space for full shift (bias*2), mantissa, sign and jbit
+    var fxdMantissa = [
+      if (isNormal()) LogicValue.one else LogicValue.zero,
+      mantissa
+    ].swizzle().zeroExtend(bias * 2 + mantissaWidth + 2);
+    fxdMantissa = sign == LogicValue.one
+        ? ~fxdMantissa + 1
+        : fxdMantissa; // make sure the sign is correct
+    final shift = exponent.toInt() - 1;
+    final shiftedFxdMantissa =
+        exponent.toInt() > 0 ? fxdMantissa << shift : fxdMantissa;
```

Comment:
Are we shifting out the unbiased exponent here? I would think you would need to debias the exponent but maybe I'm missing something further here?

Reply:
I answered poorly.  yes, we are shifting the mantissa by the unbiased exponent.  That way, the tiniest numbers are all the way to the right (from mantissa-1 to 0).  Then the largest numbers are shifted as far left as we can so the mantissa sits at the left edge down to left edge - mantissa-width.  that makes the fixed-point around 2*bias + mantissa-width wide.